### PR TITLE
Implement EV Transactions

### LINKED ISSUE #92: EV Transactions
Add transactions and helper functions to manipulate EVs of a Pokemon.

### PR DESCRIPTION
**Pull request by adapap**
EV Transactions take a target, stat (e.g. STAT_DEF), and amount.

~Edit: Blocked by #150 for now.~
Closes #92

### CODE REVIEW

File: items.go [hunk 1]
```diff
@@ -99,7 +101,21 @@ func (p *Pokemon) UseItem(i *Item) []Transaction {
...
+		t = append(t, HealTransaction{Target: p, Amount: 20})
 	}
 	return t
 }
+
+// Uses a vitamin which affects EVs and friendship
+func (p *Pokemon) UseVitamins(i *Item) (t []Transaction) {
+	// https://bulbapedia.bulbagarden.net/wiki/Friendship#In_Generation_IV
+	switch i.ID {
+	case ITEM_CALCIUM:
```
Comment (carpenterd777) [inline items.go:113]: do we want to implement vitamins in battle? in the original games, you cannot use vitamins in battle. (see https://bulbapedia.bulbagarden.net/wiki/Protein for an example) the only case in which a Pokemon can usually gain EVs in battle is by knocking out other Pokemon, though EV yield has not yet been generated from the data.

Comment (adapap) [reply items.go:113]: is there a list of battle-usable items? i thought I filtered the generated data to only allow those, but maybe it needs more filtering.

Comment (carpenterd777) [reply items.go:113]: yeah, there should be one in the miscellaneous section of the wiki

File: battle_test.go [hunk 2]
```diff
@@ -22,11 +22,13 @@ func (a dumbAgent) Act(ctx *BattleContext) Turn {
...
+type itemAgent struct {
+	item int
+}
 
-// Always uses a potion on first Pokemon
-func (a healAgent) Act(ctx *BattleContext) Turn {
-	item := GetItem(ITEM_POTION)
+// Always uses the specified item on the first Pokemon
+func (a itemAgent) Act(ctx *BattleContext) Turn {
+	item := GetItem(a.item)
```
Comment (dyc3) [inline battle_test.go:31]: Probably shouldn't change this, `healAgent` is used in other tests.

Use `RcAgent` instead

**Review by dyc3 — CHANGES_REQUESTED** [review]

**Comment by dyc3** [comment]
I think this PR is too out of date by now to merge in smoothly, and it's pretty small. Can we close it? @adapap

**Review by dyc3 — APPROVED** [review]

**Pull request merged**